samples/cpp/benchmark_app/remote_tensors_filling.cpp

```diff
@@ -189,3 +195,51 @@ std::map<std::string, ov::Tensor> get_remote_output_tensors(const ov::CompiledMo
 #endif
 }
 }  // namespace gpu
+
+namespace npu {
+
+std::map<std::string, ov::TensorVector> get_remote_input_tensors(
+    const std::map<std::string, std::vector<std::string>>& inputFiles,
+    const std::vector<benchmark_app::InputsInfo>& app_inputs_info,
+    const ov::CompiledModel& compiledModel,
+    size_t num_requests) {
+    slog::info << "Device memory will be used for input blobs" << slog::endl;
+    if (inputFiles.size()) {
+        slog::warn << "Device memory supports only random data at this moment, input images will be ignored"
+                   << slog::endl;
+    }
+
+    std::map<std::string, ov::TensorVector> remoteTensors;
+    auto context = compiledModel.get_context();
+    auto& zeroContext = static_cast<ov::intel_npu::level_zero::ZeroContext&>(context);
+
+    for (size_t i = 0; i < num_requests; i++) {
+        for (auto& inputs_info : app_inputs_info) {
+            for (auto& input : inputs_info) {
+                // Fill random
+                slog::info << "Prepare remote blob for input '" << input.first << "' with random values ("
+                           << std::string((input.second.is_image() ? "image" : "some binary data")) << " is expected)"
+                           << slog::endl;
+
+                auto tensor = zeroContext.create_l0_host_tensor(input.second.type,
+                                                                input.second.dataShape,
+                                                                ov::intel_npu::TensorType::INPUT);
+                remoteTensors[input.first].push_back(tensor);
+
+                if (inputFiles.empty()) {
+                    // Filling in random data
+                    auto elementsNum = std::accumulate(begin(input.second.dataShape),
+                                                       end(input.second.dataShape),
+                                                       1,
+                                                       std::multiplies<size_t>());
+
+                    fill_buffer(tensor.get(), elementsNum, input.second.type);
+                } else {
+                    // TODO: add filling with real image data
```

Comment:
It seems this line is a better place to report `Device memory supports only random data at this moment, input images will be ignored` because it comes right from a place where the implementation should be. That also eliminates TODO comment.

Convert the warning to exception to trigger an error. It's easy to fix the error: remove input files. At the same time a user won't miss in the logs that the inputs are ignored.

Please add `NPU` to the error message because it's NPU specific limitation